src/mockserver.js

```diff
@@ -0,0 +1,106 @@
+import fs from 'fs';
+
+import { logger } from '@user-office-software/duo-logger';
+import { mockServerClient } from 'mockserver-client';
+
+async function mockserver() {
+    const respondToPostRequest = function (request) {
+        if (request.method !== 'POST') {
+            return;
+        }
+        let file;
+        const name = String(request.body.xml);
+        const match = name.match('<tns:(.*?)>');
+        const method = match[1];
+
+        if (
+            method === 'getBasicPersonDetailsFromUserNumber' ||
+            method === 'getSearchableBasicPeopleDetailsFromUserNumbers'
+        ) {
+            let regexp = '';
+            method === 'getSearchableBasicPeopleDetailsFromUserNumbers'
+                ? (regexp = '<UserNumbers>(.*?)<')
+                : (regexp = '<UserNumber>(.*?)<');
+            const match = name.match(regexp);
+
+            file = JSON.parse(
+                fs.readFileSync(
+                    'src/responses/user/' + method + '/' + match[1] + '.txt',
+                    'utf8'
+                )
+            );
```

Comment:
We've reduced duplication by moving the `return` to the end of the function, but all `if` statements still include this block where only the filename changes. To reduce duplication even more, we should replace the `file` variable with a `responsePath` variable, assign it to the right path in the `if` statement, then load the file from the `responsePath` variable at the end of the function.